R/env.R

```diff
@@ -311,176 +367,220 @@ mut_parent_env <- function(env, new_env) {
 }
 
 
-#' Assign objects to an environment.
-#'
-#' These functions create bindings in the specified environment. The
-#' bindings are supplied as pairs of names and values, either directly
-#' (`env_assign()`), in dots (`env_define()`), or from a dictionary
-#' (`env_bind()`). See [is_dictionaryish()] for the definition of a
-#' dictionary.
-#'
-#' These functions operate by side effect. For example, if you assign
-#' bindings to a closure function, the environment of the function is
-#' modified in place.
-#'
-#' @inheritParams get_env
-#' @param nm The name of the binding.
-#' @param x The value of the binding.
-#' @param ... Pairs of unique names and R objects used to define new
-#'   bindings.
-#' @return The input object `env`, with its associated environment
+#' Bind symbols to objects in an environment
+#'
+#' @description
+#'
+#' These functions create bindings in an environment. These bindings
+#' are supplied through `...` as pairs of names and values or
+#' expressions. `env_bind()` is equivalent to evaluating a `<-`
+#' expression within the given environment, e.g. `with_env(env, foo <-
+#' "bar")`. This function should take care of the majority of use
```

Comment:
Maybe: "Note this is not the same as `assign()`, which evaluates the value in the current environment, not the target environment"